packages/react/src/components/List/List.tsx

```diff
@@ -135,7 +145,15 @@ class List extends AutoControlledComponent<WithAsProp<ListProps>, ListState> {
   }
 
   renderComponent({ ElementType, classes, accessibility, unhandledProps }) {
-    const { children } = this.props
+    const { children, container, items } = this.props
+
+    const hasContent = childrenExist(children) || (items && items.length > 0)
+    const containerContent = childrenExist(children) ? children : this.renderItems()
+
+    const maybeWrappedContent =
```

Comment:
😥😥😥  I don't like introducing a new idea for a slot that is not a wrapper, but it is wrapping the content, but I don't have any better idea of how this can be achieved with the current implementation. Can we implement this as a real wrapper slot (around the list), so that we can at least reuse the logic we have in other components (MenuItem, Input etc...), or it has to be between the list and the items. It looks strange that the generated structure would be
```
<ul>
  <divl>
    <li />
    <li />
...
```

Reply:
I wanted to avoid having wrapper around the whole list. This causes the problem of not being sure if the unhandled props are passed to the first or second element. This way, the list is always top-most element.

Yes, DOM structure is wrong in this case. But I think we will need to switch to `<div>s` everywhere (even in menu and list) with a propper aria role just because of the custom scrollbar.

Reply:
What about `wrap` prop instead? It will be not so flexible, but without `Box`es and additional DOM elements :)

```tsx
  renderComponent({ ElementType, classes, accessibility, unhandledProps }) {
    const { children, items, wrap } = this.props
    const hasContent = childrenExist(children) || (items && items.length > 0)

    return (
      <ElementType
        {...accessibility.attributes.root}
        {...rtlTextContainer.getAttributes({ forElements: [children] })}
        {...unhandledProps}
        {...applyAccessibilityKeyHandlers(accessibility.keyHandlers.root, unhandledProps)}
        className={classes.root}
      >
        {hasContent && wrap(childrenExist(children) ? children : this.renderItems())}
      </ElementType>
    )
  }
```

And then it can be used as:

```jsx
      <Dropdown
        search
        items={items}
        list={{ wrap: (children) => <Scrollbars style={{ height: '20rem' }}>{children}</Scrollbars> }}
      />
```